grails-app/controllers/org/pih/warehouse/RoleInterceptor.groovy

```diff
@@ -140,9 +141,12 @@ class RoleInterceptor {
     }
 
     static Boolean needSuperuser(controllerName, actionName) {
-        superuserControllers?.contains(controllerName) || superuserActions[controllerName]?.contains(actionName) || superuserActions['*'].any {
-            actionName?.startsWith(it)
-        }
+        (superuserActions[controllerName]?.contains("*")
+            || superuserControllers?.contains(controllerName)
+            || superuserActions[controllerName]?.contains(actionName)
+            || superuserActions['*'].any {
+                actionName?.startsWith(it)
+            })
```

Comment:
hmmm, do we have a styling rule about the placing of the logical operators?
I mean whether we should place them in the way you did, or:
```suggestion
        (superuserActions[controllerName]?.contains("*") ||
            superuserControllers?.contains(controllerName) ||
            superuserActions[controllerName]?.contains(actionName) ||
            superuserActions['*'].any {
                actionName?.startsWith(it)
            })
```

Reply:
According to Java coding conventions, a break should happen **before** the operator: https://www.oracle.com/java/technologies/javase/codeconventions-indentation.html

I think I've raised this topic recently in one of the PRs, that we should discuss it and document it in our coding conventions.